TPTP/Prover/SPASS/ProofParser.hs

```diff
@@ -43,7 +46,32 @@ parseOutput namedGoal = foldl parseIt (Nothing, [], False, midnight)
                 fromMaybe midnight $ parseTimeOfDay $
                   takeWhile (\ c -> isDigit c || elem c ".:") $ trimLeft s
               _ -> timeUsed
-          )
+          , spass_proof_passed ||  ("Here is a proof " `isPrefixOf` line)
+          , case (spass_proof_passed, parsedProofLine) of
+              (True, Right (clId, rule, _, clSntc)) -> [(clId, Right (show clId,clSntc)),(-clId-1,Left rule)]++clNodes
+              _ -> clNodes
+          , case (spass_proof_passed, parsedProofLine) of
+              (True, Right (clId, _, parents, _)) -> [(p, -clId-1, i) | (i,p) <- zip [0..] parents] ++ (-clId-1,clId,-1):clEdges
+              _ -> clEdges
+          ) where parsedProofLine = runParser proofLine () "" line
+
+
+proofLine :: Parser (Int, String, [Int], String)
+proofLine = do
+ clauseId <- decimal
+ char '['
+ decimal -- What's this?
```

Comment:
this is the split level, see "OUTPUT" at https://webspass.spass-prover.org/help/options.html

Reply:
here more about the notion of split level: https://core.ac.uk/download/pdf/205222916.pdf

Reply:
I guess I'd reverse-engineered the output before I found the documentation you linked and didn't bother looking at the OUTPUT section after I finally found it in the context of TPTP output. Thanks for pointing this out!

It seems to me that the labelling and therefore the split level is just bookkeeping to enable non-chronological backtracking. As far as I can tell (which, admittedly, is not very far), the proof graph makes it apparent at which levels the splits happened, since only the ones that contributed to finding the refutation are recorded in the output proof. I'd therefore conclude that we don't need to include this information in a proof graph.

What do you think?